pkg/tfgen/installation_docs.go

```diff
@@ -70,15 +70,20 @@ func writeFrontMatter(providerName string) string {
 		title)
 }
 
-func getBodyAndTitle(content string) (string, string) {
-	// The first header in `index.md` is the package name, of the format `# Foo Provider`.
+func stripUpstreamFrontMatter(content string) string {
```

Comment:
There is already a `trimFrontMatter` function in `pkg/tfgen/docs.go`. Can we just use that?

Also, it looks like this function doesn't just strip frontmatter, it also removes the first H1 encountered. Can we do this with goldmark instead of manual parsing?

For example, this is a valid markdown H1:

```md
H1
===
```

We don't need to write code to understand `===`.

Reply:
Thanks for pointing out that function! I....had to adjust the trim index at the end because it would deliver a little `-` otherwise - but I added a test that should keep it behaving.